src/main/java/io/jenkins/plugins/checks/api/ChecksDetails.java

```diff
@@ -191,9 +192,9 @@ public ChecksDetailsBuilder withOutput(final Output output) {
          * @return this builder
          * @throws NullPointerException if the {@code actions} is null
          */
-        public ChecksDetailsBuilder withActions(final List<Action> actions) {
+        public ChecksDetailsBuilder withActions(final List<ChecksAction> actions) {
             requireNonNull(actions);
-            this.actions = Collections.unmodifiableList(actions);
+            this.actions = Collections.unmodifiableList(new ArrayList<>(actions));
```

Comment:
unmodifiable list returns a copy of the list anyway, no need to double copy
```suggestion
            this.actions = Collections.unmodifiableList(actions);
```